src/main/java/Duke.java

```diff
@@ -1,10 +1,431 @@
+
+                tasks.get(taskIndex).setTaskDone(true);
+                
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
+
+                System.out.println("Nice! I've marked this task as done: \n" + taskMark + doneMark +
+                        tasks.get(taskIndex).getTaskTitle());
+
+                String newEntry = parseTaskToString(tasks);
+                writeDisk(newEntry);
+            } catch (InvalidListIndexException | NumberFormatException e) {
+                System.out.println("Error - List index given invalid. Please check again.");
+            } catch (IOException e) {
+                System.out.println("Error when saving file: " + e.getMessage());
+            }
+            break;
+            
+            case "unmark":
+            try {
+                if (inputTaskIndex > tasks.size() || inputTaskIndex < 1) {
+                    throw new InvalidListIndexException();
+                }
+
+                tasks.get(taskIndex).setTaskDone(false);
+
+
+                String taskMark = getTaskMark(inLine);
+                String doneMark;
+                if (tasks.get(taskIndex).getTaskDone()){
+                    doneMark = "[X]";
+                } else{
+                    doneMark = "[ ]";
+                }
+
+                System.out.println("Nice! I've marked this task as not done: \n" + taskMark + doneMark +
+                        tasks.get(taskIndex).getTaskTitle());
+
+                String newEntry = parseTaskToString(tasks);
+                writeDisk(newEntry);
+            } catch (InvalidListIndexException | NumberFormatException e) {
+                System.out.println("Error - List index given invalid. Please check again.");
+            } catch (IOException e) {
+                System.out.println("Error when saving file: " + e.getMessage());
+            }
+            break;
+            
+            case "todo":
+            try {
+                if (inLine.length <= 1) {
+                    throw new InvalidTodoException();
+                }
+
+                tasks.add(new ToDo(inData.substring(5)));
+
+                System.out.println("Got it. I've added this task: \n" + "[T][ ] "
+                        + inData.substring(5) ); //////////magic number problem
+                System.out.println("Now you have " + tasks.size() +  " tasks in the list.");
+
+                String newEntry = parseTaskToString(tasks);
+                writeDisk(newEntry);
+            } catch (InvalidTodoException e) {
+                System.out.println("Error - Please input a description for your To do. ");
+            } catch (IOException e) {
+                System.out.println("Error when saving file: " + e.getMessage());
+            }
+            break;
```

Comment:
You should consider indenting this block of code one level further, as stated in the coding standard.